### PR TITLE
Option to turn off filenames and line numbers in requirements.txt

### PR DESCRIPTION
**Pull request by bganglia**
This addresses one suggestion from #65 

It adds an option ("-m" or "--minimal") that disables printing of the files and line numbers where each module is printed. By default, the files and line numbers are still printed.

### CODE REVIEW

File: pigar/__main__.py [hunk 1]
```diff
@@ -197,16 +203,17 @@ def extract_reqs(self):
...
 
-    def _write_reqs(self, reqs):
+    def _write_reqs(self, reqs, minimal=False):
         print(Color.BLUE('Writing requirements to "{0}"'.format(
             self._save_path)))
         with open(self._save_path, 'w+') as f:
             f.write('# Requirements automatically generated by pigar.\n'
                     '# https://github.com/damnever/pigar\n')
             for k, v in reqs.sorted_items():
                 f.write('\n')
```
Comment (damnever) [inline pigar/__main__.py]: The empty line is for comments to improve readability, I do not think leaving it here is a good idea.

File: pigar/cmd.py [hunk 2]
```diff
@@ -84,13 +84,19 @@ def parse_args(args=None):
         default=['=='],
         help='The comparison operator for versions, alternatives: [==, ~=, >=]'
     )
+    parser.add_argument(
+        '-m', '--minimal',
+        dest='minimal',
+        action='store_true',
```
[the remark this answers is not in the document]
Comment (bganglia) [reply pigar/cmd.py]: @damnever Thanks. I changed the name and added a help message

File: pigar/__main__.py (hunk 1 above)
Comment (bganglia) [reply pigar/__main__.py]: @damnever Good point. I added a comment to fill this line.

Comment (damnever) [reply pigar/__main__.py]: What is the content?

File: pigar/cmd.py [hunk 3]
```diff
@@ -84,13 +84,21 @@ def parse_args(args=None):
         default=['=='],
         help='The comparison operator for versions, alternatives: [==, ~=, >=]'
     )
+    parser.add_argument(
+        '--without-referenced-commits',
```
Comment (damnever) [inline pigar/cmd.py]: typo 😅

File: pigar/__main__.py (hunk 1 above)
Comment (damnever) [reply pigar/__main__.py]: What is the content?

File: pigar/cmd.py (hunk 3 above)
Comment (damnever) [inline pigar/cmd.py]: typo 😅

File: pigar/__main__.py (hunk 1 above)
Comment (bganglia) [reply pigar/__main__.py]: @damnever the content is just """Write requirements to a file"""

Comment (damnever) [reply pigar/__main__.py]: em.. the empty line is `f.write('\n')`.

Comment (bganglia) [reply pigar/__main__.py]: Oh, now I see. I will not print the empty line in the requirements.txt if the user has chosen not to show referenced files.

File: pigar/cmd.py (hunk 3 above)
Comment (bganglia) [reply pigar/cmd.py]: @damnever I am sorry I do not see the typo. Did you fix it already?

File: pigar/__main__.py [hunk 4]
```diff
@@ -197,16 +203,18 @@ def extract_reqs(self):
...
         return reqs, try_imports, guess
 
-    def _write_reqs(self, reqs):
+    def _write_reqs(self, reqs, minimal=False):
+        """Write requirements to a file"""
         print(Color.BLUE('Writing requirements to "{0}"'.format(
             self._save_path)))
         with open(self._save_path, 'w+') as f:
             f.write('# Requirements automatically generated by pigar.\n'
                     '# https://github.com/damnever/pigar\n')
```
Comment (damnever) [inline pigar/__main__.py]: I think here needs an extra `f.write('\n')` if `not minimal`.

File: pigar/cmd.py (hunk 3 above)
Comment (damnever) [reply pigar/cmd.py]: `commits` -> `comments`?

Comment (bganglia) [reply pigar/cmd.py]: @damnever Thank you, I see now. I will fix it.

File: pigar/__main__.py (hunk 4 above)
Comment (bganglia) [reply pigar/__main__.py]: That makes sense. I will add it.

**Comment by bganglia** [comment]
@damnever I just figured out why the unit tests are failing. They do not expect parse_args to return an extra item (minimal) at the end of the list. I will modify test_cmd.py to include this.

**Comment by bganglia** [comment]
@damnever I notice there are individual tests for many of the options. Should there be another for this one?

File: pigar/cmd.py [hunk 5]
```diff
@@ -84,13 +84,21 @@ def parse_args(args=None):
         default=['=='],
         help='The comparison operator for versions, alternatives: [==, ~=, >=]'
     )
+    parser.add_argument(
+        '--without-referenced-comments',
+        dest='minimal',
+        action='store_true',
+        help='omit requirements.txt comments showing the file and line of '
```
Comment (damnever) [inline pigar/cmd.py]: The other help messages all start with an upper case, we need to keep it consistent here as well.

Comment (bganglia) [reply pigar/cmd.py]: @damnever Good point. I will change that right away

File: pigar/__main__.py [hunk 6]
```diff
@@ -197,16 +203,20 @@ def extract_reqs(self):
...
-            for k, v in reqs.sorted_items():
+            if not minimal:
```
Comment (damnever) [inline pigar/__main__.py]: Have you ever test it?

Comment (bganglia) [reply pigar/__main__.py]: @damnever I think I see. I changed the code so there is one blank line after the first comment, regardless of whether the option is used.

with --without-referenced-comments
```
# Requirements automatically generated by pigar.
# https://github.com/damnever/pigar

numpy == 1.18.1
opencv-contrib-python == 4.1.2.30
...
```

without the option
```
# Requirements automatically generated by pigar.
# https://github.com/damnever/pigar

# examples/Plant-Leaf-Disease-Detection/neural_network.py: 9
numpy == 1.18.1
...
```
Is that better?

**Comment by bganglia** [comment]
Perhaps I could change the variable "minimal" to "no_comments" or "without_comments". Would that be better?

**Pull request merged**